utils/_context/_scenarios/stream_processing_offload.py

```diff
@@ -0,0 +1,131 @@
+import pytest
+
+from utils._context.containers import (
+    DummyServerContainer,
+    StreamProcessingOffloadContainer,
+    HAProxyContainer,
+    AgentContainer,
+)
+from utils import interfaces
+from utils.interfaces._core import ProxyBasedInterfaceValidator
+
+from utils._logger import logger
+
+from .core import scenario_groups
+from .endtoend import DockerScenario
+
+
+class StreamProcessingOffloadScenario(DockerScenario):
+    def __init__(
+        self,
+        name: str,
+        doc: str,
+        *,
+        stream_processing_offload_env: dict[str, str | None] | None = None,
+        stream_processing_offload_volumes: dict[str, dict[str, str]] | None = None,
+        rc_api_enabled: bool = False,
+    ) -> None:
+        super().__init__(
+            name,
+            doc=doc,
+            github_workflow="streamprocessingoffload",
+            scenario_groups=[
+                scenario_groups.end_to_end,
+                scenario_groups.stream_processing_offload,
+                scenario_groups.all,
+            ],
+            use_proxy=True,
+            rc_api_enabled=rc_api_enabled,
+        )
+
+        self._agent_container = AgentContainer(self.host_log_folder)
+        self._stream_processing_offload_container = StreamProcessingOffloadContainer(
+            self.host_log_folder,
+            env=stream_processing_offload_env,
+            volumes=stream_processing_offload_volumes,
+        )
+        self._haproxy_container = HAProxyContainer(self.host_log_folder)
+        self._http_app_container = DummyServerContainer(self.host_log_folder)
+
+        self._agent_container.depends_on.append(self.proxy_container)
+        self._stream_processing_offload_container.depends_on.append(self.proxy_container)
+        # Ensure HAProxy starts after SPOE agent and the HTTP app so DNS resolves
+        self._haproxy_container.depends_on.append(self._stream_processing_offload_container)
+        self._haproxy_container.depends_on.append(self._http_app_container)
+
+        self._required_containers.append(self._agent_container)
+        self._required_containers.append(self._stream_processing_offload_container)
+        self._required_containers.append(self._haproxy_container)
+        self._required_containers.append(self._http_app_container)
+
+    def configure(self, config: pytest.Config):
```

Comment:
you'll need to rebase, and change the signature of `configure`, I just changed it.